shared/js/gzip/gzip_worker.js

```diff
@@ -0,0 +1,23 @@
+/* global Zlib */
+'use strict';
+
+importScripts('gzip.min.js');
+
+self.onmessage = function(message) {
+  var cmd = message.data.cmd;
+
+  if (cmd === 'gzip') {
+
+    // gzip function requires Uint8Array
+    var textEncoder = new TextEncoder('utf8');
+    var payload = message.data.payload;
+    var uint8Data = textEncoder.encode(payload);
+
+    var gzip = new Zlib.Gzip(uint8Data);
+    var compressed = gzip.compress();
```

Comment:
Marshall makes a good point. You should have a try{} block around the two lines above and postMessage() some kind of error from the catch clause. Otherwise, gzip.js is just waiting 10 seconds or whatever for a timeout.